go-app-ussd_nurse.js

```diff
@@ -72,6 +82,82 @@ go.app = function() {
             });
         };
 
+        self.jembi_nc_clinic_validate = function (im, clinic_code) {
+            var params = {
+                'criteria': 'value:' + clinic_code
+            };
+            return self
+            .jembi_json_api_call('get', params, null, 'NCfacilityCheck', im);
+        };
+
+        self.validate_nc_clinic_code = function(im, clinic_code) {
+            if (!utils.check_valid_number(clinic_code) ||
+                clinic_code.length !== 6) {
+                return Q()
+                    .then(function() {
+                        return false;
+                    });
+            } else {
+                return self
+                .jembi_nc_clinic_validate(im, clinic_code)
+                .then(function(json_result) {
+                    var rows = json_result.data.rows;
+                    if (rows.length === 0) {
+                        return false;
+                    } else {
+                        return rows[0][2];
+                    }
+                });
+            }
+        };
+
+        self.jembi_json_api_call = function(method, params, payload, endpoint, im) {
+            var http = new JsonApi(im, {
+                auth: {
+                    username: im.config.jembi.username,
+                    password: im.config.jembi.password
+                }
+            });
+            switch(method) {
+                case "post":
+                    return http.post(im.config.jembi.url_json + endpoint, {
+                        data: payload
+                    });
+                case "get":
+                    return http.get(im.config.jembi.url_json + endpoint, {
+                        params: params
+                    });
+            }
+        },
+
+        // temporary - TODO: adapt IdentityStore class in seed-jsbox-utils (#15) to have optin function
+        self.optin_identity = function(identity) {
+            var http = new JsonApi(self.im, {});
+            http.defaults.headers.Authorization = ['Token ' + self.im.config.services.identity_store.token];
+
+            var endpoint = 'optin/';
+            var url = self.im.config.services.identity_store.url + endpoint;
+
+            return http.post(url, {data: identity})
+                //.then(this.log_service_call('POST', url, identity, null))
+                .then(function(response) {
+                    return response.data;
+                });
+        },
+
+    // REGISTRATION FINISHED SMS HANDLING
+
+        self.send_registration_thanks = function(msisdn) {
```

Comment:
@imsickofmaps thought we might want to... seed-jsbox-utils will also consequently have to be updated.  Currently MessageSender only saves inbound messages...